packages/edit-site/src/components/page-pages/index.js

```diff
@@ -0,0 +1,190 @@
+	const [ globalFilter, setGlobalFilter ] = useState( '' );
+	const [ paginationInfo, setPaginationInfo ] = useState();
+	const [ { pageIndex, pageSize }, setPagination ] = useState( {
+		pageIndex: 0,
+		pageSize: PAGE_SIZE_VALUES[ 0 ],
+	} );
+	// Request post statuses to get the proper labels.
+	const [ postStatuses, setPostStatuses ] = useState( EMPTY_ARRAY );
+	useEffect( () => {
+		apiFetch( {
+			path: '/wp/v2/statuses',
+		} ).then( setPostStatuses );
+	}, [] );
+
+	// TODO: probably memo other objects passed as state(ex:https://tanstack.com/table/v8/docs/examples/react/pagination-controlled).
+	const pagination = useMemo(
+		() => ( { pageIndex, pageSize } ),
+		[ pageIndex, pageSize ]
+	);
+	const [ sorting, setSorting ] = useState( [
+		{ order: 'desc', orderby: 'date' },
+	] );
+	const queryArgs = useMemo(
+		() => ( {
+			per_page: pageSize,
+			page: pageIndex + 1, // tanstack starts from zero.
+			_embed: 'author',
+			order: sorting[ 0 ]?.desc ? 'desc' : 'asc',
+			orderby: sorting[ 0 ]?.id,
+			search: globalFilter,
+			status: [ 'publish', 'draft' ],
+		} ),
+		[
+			globalFilter,
+			sorting[ 0 ]?.id,
+			sorting[ 0 ]?.desc,
+			pageSize,
+			pageIndex,
+			reset,
+		]
+	);
+	const { records, isResolving: isLoading } = useEntityRecords(
+		'postType',
+		'page',
+		queryArgs
+	);
+	useEffect( () => {
+		// Make extra request to handle controlled pagination.
+		apiFetch( {
+			path: addQueryArgs( '/wp/v2/pages', {
+				...queryArgs,
+				_fields: 'id',
+			} ),
+			method: 'HEAD',
+			parse: false,
+		} ).then( ( res ) => {
+			const totalPages = parseInt( res.headers.get( 'X-WP-TotalPages' ) );
+			const totalItems = parseInt( res.headers.get( 'X-WP-Total' ) );
+			setPaginationInfo( {
+				totalPages,
+				totalItems,
+			} );
+		} );
+		// Status should not make extra request if already did..
+	}, [ globalFilter, pageSize, reset ] );
+
+	const fields = useMemo(
+		() => [
+			{
+				header: __( 'Title' ),
+				id: 'title',
+				accessorFn: ( page ) => page.title?.rendered || page.slug,
+				cell: ( props ) => {
+					const page = props.row.original;
+					return (
+						<VStack spacing={ 1 }>
+							<Heading as="h3" level={ 5 }>
+								<Link
+									params={ {
+										postId: page.id,
+										postType: page.type,
+										canvas: 'edit',
+									} }
+								>
+									{ decodeEntities( props.getValue() ) }
+								</Link>
+							</Heading>
+						</VStack>
+					);
+				},
+				maxWidth: 400,
+				sortingFn: 'alphanumeric',
+				enableHiding: false,
+			},
+			{
+				header: __( 'Author' ),
+				id: 'author',
+				accessorFn: ( page ) => page._embedded?.author[ 0 ]?.name,
+				cell: ( props ) => {
+					const author = props.row.original._embedded?.author[ 0 ];
+					return (
+						<a href={ `user-edit.php?user_id=${ author.id }` }>
+							{ author.name }
+						</a>
+					);
+				},
+			},
+			{
+				header: 'Status',
+				id: 'status',
+				cell: ( props ) =>
```

Comment:
Created a follow-up at https://github.com/WordPress/gutenberg/pull/55196 It seems we should use `accessorFn` instead of `cell` to render this value.